src/components/hooks/useThumbnailCrop.tsx

```diff
@@ -0,0 +1,91 @@
+import { RefObject, useEffect, useRef, useState } from 'react';
```

Comment:
I think this one could be in ui! To be reused in account potentially. Actually probably the whole cropping component.